models/intermediate/int_zendesk__updates.sql

```diff
@@ -27,23 +61,50 @@ with ticket_history as (
     select
         source_relation,
         ticket_id,
-        cast('comment' as {{ dbt.type_string() }}) as field_name,
+        cast('comment - not chat' as {{ dbt.type_string() }}) as field_name,
         body as value,
         is_public,
         user_id,
         created_at as valid_starting_at,
         lead(created_at) over (partition by source_relation, ticket_id order by created_at) as valid_ending_at
     from ticket_comment
 
+{% if var('using_ticket_chat', False) %}
+    union all
+
+    select
+        source_relation,
+        ticket_id,
+        cast('comment - chat' as {{ dbt.type_string() }}) as field_name,
+        message as value,
+        true as is_public,
+        actor_id as user_id,
+        created_at as valid_starting_at,
+        lead(created_at) over (partition by source_relation, ticket_id order by created_at) as valid_ending_at
+    from ticket_chat_join
+{% endif %}
+
 ), final as (
     select
-        updates_union.*,
+        updates_union.source_relation,
+        updates_union.ticket_id,
+        case 
+            when updates_union.field_name in ('comment - chat', 'comment - not chat') then 'comment' 
+        else updates_union.field_name end as field_name,
+        updates_union.value,
+        updates_union.is_public,
+        updates_union.user_id,
+        updates_union.valid_starting_at,
+        updates_union.valid_ending_at,
         tickets.created_at as ticket_created_date
     from updates_union
 
     left join tickets
         on tickets.ticket_id = updates_union.ticket_id
         and tickets.source_relation = updates_union.source_relation
+
+    where not (updates_union.field_name = 'comment - not chat' and lower(tickets.created_channel) in ('chat', 'native_messaging')) -- Only keep chat comments for chat/native_messaging tickets
```

Comment:
Why not just filter out the `chat` and `native_messaging` directly in the query on line 37? Even if a customer doesn't have the ticket_chat enabled, we should filter these out to make sure we aren't reporting incorrect metrics for those sla's.

Reply:
If we do the above, then the `comment - not chat` and `comment - chat` strings won't be necessary anymore.

Reply:
Yeah so this filter is directed at the comment-batches, which we can isolate based on having the `comment - not chat` flag AND a `chat` or `native_messaging` `created_channel` value.